### DISCUSSION TITLE
[asyncpg] Accessing eagerly-loaded empty relationship triggers new lazy-load, raising error

### DISCUSSION THREAD

**Discussion by theo-brown** [sqlalchemy/sqlalchemy#7196] [Usage Questions] [answered]
I am using sqlalchemy + asyncpg, and 'selectin' eager loading.

I have Person items that have one-to-many relationships with Friends.

I insert a Person into my database, with no related Friend entries. If in the same session I try and get that Person from the database, I can access their static (non-relationship) columns fine, but cannot access the friends relationship.

I think trying to access person.friends is triggering a lazy load, despite it being enforced previously as a selectin load. Why is this? How can I avoid it?
```python
# Create the ORM model
class Person(Base):
    __tablename__ = 'items'
    id_ = Column(POSTGRES_UUID(as_uuid=True), primary_key=True)
    name = Column(String(32))
    friends = relationship('Friend', lazy='selectin')

# Create an instance
person_id = uuid4()
person = Person(id_=person_id, name='Alice') # Note that this Person's friends are not set

# Add to database
async with AsyncSession(engine, expire_on_commit=False) as session:
    try:
        session.begin()
        session.add(person)
        await session.commit()
    except:
        await session.rollback()
        raise
    # Get the added person from the database
    created_person = await session.get(person, person_id)
    print(created_person.id_) # Works fine
    print(created_person.friends) # Raises sqlalchemy.exc.MissingGreenlet
```
Moving the section under `# Get the added person from the database` to a new AsyncSession context means that it works without raising an error. Do I need to flush() or something?

**Comment by zzzeek** [comment, marked as answer]
this is the same issue at #7176.  use populate_existing with session.get().

**Reply by CaselIT** [reply]
Not really sure what we should change here.

Maybe the loader could intercept that error and suggest populate existing?
